langchain4j-agentic/src/main/java/dev/langchain4j/agentic/workflow/impl/ConditionalPlanner.java

```diff
@@ -0,0 +1,29 @@
+package dev.langchain4j.agentic.workflow.impl;
+
+import dev.langchain4j.agentic.planner.Action;
+import dev.langchain4j.agentic.planner.AgentInstance;
+import dev.langchain4j.agentic.planner.Planner;
+import dev.langchain4j.agentic.scope.AgentExecution;
+import dev.langchain4j.agentic.scope.AgenticScope;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+public record ConditionalPlanner(List<ConditionalAgent> conditionalAgents) implements Planner {
+
+    record ConditionalAgent(Predicate<AgenticScope> condition, List<AgentInstance> agentInstances) { }
+
+    @Override
+    public Action firstAction(AgenticScope agenticScope) {
+        List<AgentInstance> agentsToCall = conditionalAgents.stream()
+                .filter(conditionalAgent -> conditionalAgent.condition.test(agenticScope))
+                .flatMap(conditionalAgent -> conditionalAgent.agentInstances.stream())
+                .toList();
```

Comment:
Interesting! When thinking of conditional planning, I was thinking of getting just one action to be triggered, but indeed, that's a sub-list of agents! Neat & powerful!